packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart

```diff
@@ -0,0 +1,388 @@
+    LogicalKeySet(LogicalKeyboardKey.alt, LogicalKeyboardKey.arrowUp): const MoveSelectionToStartTextIntent(),
+    LogicalKeySet(LogicalKeyboardKey.alt, LogicalKeyboardKey.shift, LogicalKeyboardKey.arrowDown): const ExpandSelectionToEndTextIntent(),
+    LogicalKeySet(LogicalKeyboardKey.alt, LogicalKeyboardKey.shift, LogicalKeyboardKey.arrowLeft): const ExpandSelectionLeftByLineTextIntent(),
+    LogicalKeySet(LogicalKeyboardKey.alt, LogicalKeyboardKey.shift, LogicalKeyboardKey.arrowRight): const ExpandSelectionRightByLineTextIntent(),
+    LogicalKeySet(LogicalKeyboardKey.alt, LogicalKeyboardKey.shift, LogicalKeyboardKey.arrowUp): const ExpandSelectionToStartTextIntent(),
+    LogicalKeySet(LogicalKeyboardKey.arrowDown): const MoveSelectionDownTextIntent(),
+    LogicalKeySet(LogicalKeyboardKey.arrowLeft): const MoveSelectionLeftTextIntent(),
+    LogicalKeySet(LogicalKeyboardKey.arrowRight): const MoveSelectionRightTextIntent(),
+    LogicalKeySet(LogicalKeyboardKey.arrowUp): const MoveSelectionUpTextIntent(),
+    LogicalKeySet(LogicalKeyboardKey.control, LogicalKeyboardKey.arrowLeft): const MoveSelectionLeftByWordTextIntent(),
+    LogicalKeySet(LogicalKeyboardKey.control, LogicalKeyboardKey.arrowRight): const MoveSelectionRightByWordTextIntent(),
+    LogicalKeySet(LogicalKeyboardKey.control, LogicalKeyboardKey.shift, LogicalKeyboardKey.arrowLeft): const ExtendSelectionLeftByWordTextIntent(),
+    LogicalKeySet(LogicalKeyboardKey.control, LogicalKeyboardKey.shift, LogicalKeyboardKey.arrowRight): const ExtendSelectionRightByWordTextIntent(),
+    LogicalKeySet(LogicalKeyboardKey.shift, LogicalKeyboardKey.arrowDown): const ExtendSelectionDownTextIntent(),
+    LogicalKeySet(LogicalKeyboardKey.shift, LogicalKeyboardKey.arrowLeft): const ExtendSelectionLeftTextIntent(),
+    LogicalKeySet(LogicalKeyboardKey.shift, LogicalKeyboardKey.arrowRight): const ExtendSelectionRightTextIntent(),
+    LogicalKeySet(LogicalKeyboardKey.shift, LogicalKeyboardKey.arrowUp): const ExtendSelectionUpTextIntent(),
+    // Meta + arrow down has no effect on text editing on this platform.
+    // End has no effect on text editing on this platform.
+    // Home has no effect on text editing on this platform.
+    // Meta + arrow left has no effect on text editing on this platform.
+    // Meta + arrow right has no effect on text editing on this platform.
+    // Meta + arrow up has no effect on text editing on this platform.
+    // Meta + shift + arrow down has no effect on text editing on this platform.
+    // Meta + shift + arrow left has no effect on text editing on this platform.
+    // Meta + shift + arrow right has no effect on text editing on this platform.
+    // Meta + shift + arrow up has no effect on text editing on this platform.
+    // Shift + end has no effect on text editing on this platform.
+    // Shift + home has no effect on text editing on this platform.
```

Comment:
Do <kbd>pageUp</kbd> and <kbd>pageDown</kbd> do anything special in a text field? They currently seem to be mapped to a `ScrollIntent`.

Reply:
Good call, I created a separate issue for that: https://github.com/flutter/flutter/issues/77611

It looks like on Mac it should scroll the TextField without moving the cursor.  On Flutter right now it will scroll an enclosing ListView but it won't scroll the TextField.